raw/fetch.py

```diff
@@ -0,0 +1,277 @@
+        'last_day_first_prev': last_day_first_prev,
+        'first_day_third_prev': first_day_third_prev,
+        'last_day_third_prev': last_day_third_prev,
+        'sl_year': sl_year,
+        'sl_month': sl_month,
+    }
+
+
+def update_config_files(directory: str, field_name: str,
+                        additional_content: str) -> None:
+    """Update the configuration files in the specified directory.
+
+    Parameters:
+        directory (str): The path to the directory containing the configuration files.
+        field_name (str): The name of the field to be updated with the new value.
+        additional_content (str): The additional content to be added under the
+                    '[selection]' section.
+    """
+    dates_data = get_previous_month_dates()
+    config_args = {
+        "first_day_first_prev": dates_data['first_day_first_prev'],
+        "last_day_first_prev": dates_data['last_day_first_prev'],
+        "first_day_third_prev": dates_data['first_day_third_prev'],
+        "last_day_third_prev": dates_data['last_day_third_prev'],
+        "sl_year": dates_data['sl_year'],
+        "sl_month": dates_data['sl_month'],
+    }
+    for filename in os.listdir(directory):
+        config_args['single_level_file'] = config_args['co_file'] = False
+        if filename.endswith(".cfg"):
+            if "sfc" in filename:
+                config_args["single_level_file"] = True
+            if "hourly" in filename:
+                config_args["co_file"] = True
+            config_file = os.path.join(directory, filename)
+            # Pass the data as keyword arguments to the new_config_file function
+            new_config_file(config_file, field_name, additional_content,
+                            config_args=config_args)
+
+
+def get_secret(api_key: str) -> dict:
+    """Retrieve the secret value from the Google Cloud Secret Manager.
+
+    Parameters:
+        api_key (str): The name or identifier of the secret in the Google
+                        Cloud Secret Manager.
+
+    Returns:
+        dict: A dictionary containing the retrieved secret data.
+    """
+    client = secretmanager.SecretManagerServiceClient()
+    response = client.access_secret_version(request={"name": api_key})
+    payload = response.payload.data.decode("UTF-8")
+    secret_dict = json.loads(payload)
+    return secret_dict
+
+
+if __name__ == "__main__":
+    for env_var in os.environ:
+        if API_KEY_PATTERN.match(env_var):
+            api_key_value = os.environ.get(env_var)
+            API_KEY_LIST.append(api_key_value)
+
+    additional_content = ""
+    for count, key in enumerate(API_KEY_LIST):
+        api_key_value = get_secret(key)
+        additional_content += f'parameters.api{count}\n\
+            api_url={api_key_value["api_url"]}\napi_key={api_key_value["api_key"]}\n\n'
+
+    current_day = datetime.date.today()
+    job_name = f"wx-dl-arco-era5-{current_day.month}-{current_day.year}"
+
+    command = (
```

Comment:
In the future, we should support running weather-dl "--async": this is where after it launches the job, it closes all logs and exists.

Reply:
I created an issue in [`weather-dl`](https://github.com/google/weather-tools/issues/373) and added TODO here.